pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -166,137 +204,153 @@ func (computeCfg *computeServiceConfig) getManagedVnets() map[string]armnetwork.
 	subscriptionIDs := []string{computeCfg.credentials.SubscriptionID}
 	tenantIDs := []string{computeCfg.credentials.TenantID}
 	locations := []string{computeCfg.credentials.region}
-
+	namespacedName := types.NamespacedName{Namespace: selector.Namespace, Name: selector.Name}
 	if filters, ok := convertSelectorToComputeQuery(selector, subscriptionIDs, tenantIDs, locations); ok {
-		key := selector.GetNamespace() + "/" + selector.GetName()
-		computeCfg.computeFilters[key] = filters
-		computeCfg.selectors[key] = selector.DeepCopy()
+		azurePluginLogger().Info("resource query filters created", "selector", namespacedName)
+		computeCfg.configChangedSelectors = append(computeCfg.configChangedSelectors, &namespacedName)
+		computeCfg.selectorConfigMap[namespacedName] = &selectorConfig{
+			namespacedName:  namespacedName,
+			instanceFilters: filters,
+			ces:             selector.DeepCopy(),
+		}
 	} else {
-		return fmt.Errorf("error creating resource query filters")
+		return fmt.Errorf("error creating resource query filters for selector: %v", namespacedName)
 	}
+
 	return nil
 }
 
 func (computeCfg *computeServiceConfig) RemoveResourceFilters(selectorNamespacedName *types.NamespacedName) {
-	delete(computeCfg.computeFilters, selectorNamespacedName.String())
-	delete(computeCfg.selectors, selectorNamespacedName.String())
+	delete(computeCfg.selectorConfigMap, *selectorNamespacedName)
+	//TODO: cannot update snapshot as updating vpcs, managed vpc ids is not possible right away.
 }
 
-func (computeCfg *computeServiceConfig) GetInternalResourceObjects(namespace string,
-	account *types.NamespacedName) map[string]*runtimev1alpha1.VirtualMachine {
-	virtualMachines := computeCfg.getCachedVirtualMachines()
+// getVirtualMachineObjects converts cached virtual machines in cloud format to internal runtimev1alpha1.VirtualMachine format.
+func (computeCfg *computeServiceConfig) getVirtualMachineObjects(accountNamespacedName *types.NamespacedName,
+	selectorNamespacedName *types.NamespacedName) map[string]*runtimev1alpha1.VirtualMachine {
+	virtualMachines := computeCfg.getCachedVirtualMachines(selectorNamespacedName)
 	vnets := computeCfg.getManagedVnets()
+
 	vmObjects := map[string]*runtimev1alpha1.VirtualMachine{}
 
 	for _, virtualMachine := range virtualMachines {
 		// build runtimev1alpha1 VirtualMachine object.
-		vmObject := computeInstanceToInternalVirtualMachineObject(virtualMachine, vnets, namespace, account, computeCfg.credentials.region)
+		vmObject := computeInstanceToInternalVirtualMachineObject(virtualMachine, vnets, selectorNamespacedName,
+			accountNamespacedName, computeCfg.credentials.region)
 		vmObjects[vmObject.Name] = vmObject
 	}
 
-	azurePluginLogger().V(1).Info("Internal resource objects", "account", computeCfg.account,
-		"VirtualMachine objects", len(vmObjects))
+	azurePluginLogger().V(1).Info("Internal resource objects", "account", computeCfg.accountNamespacedName,
```

Comment:
Do we need this log message? I beleive this information is already logged in the account manager.